test/unit/sessions/client.test.js

```diff
@@ -27,21 +27,32 @@ describe('Sessions', function() {
         });
 
         const client = this.configuration.newClient(`mongodb://${test.server.uri()}/test`);
-        client.connect(function(err, client) {
-          expect(err).to.not.exist;
-          expect(() => {
-            client.startSession();
-          }).to.throw(/Current topology does not support sessions/);
-
-          client.close(done);
-        });
+        return client
+          .connect()
+          .then(function(client) {
+            const session = client.startSession();
+            return client
+              .db()
+              .collection('t')
+              .insertOne({ a: 1 }, { session });
+          })
+          .then(() => {
+            expect.fail('Expected an error to be thrown about not supporting sessions');
```

Comment:
I always found it cumbersome with promises that you had to add an extra assertion for the case you weren't checking. Did you convert this test to promises because of personal preference, or was there some other reason?